app/gui/src/controller/graph/executed.rs

```diff
@@ -216,28 +216,30 @@ impl Handle {
         registry.clone_ref().get_type(id)
     }
 
-    /// Enter node by given node ID and method pointer.
+    /// Enter the given stack of nodes by node ID and method pointer.
     ///
-    /// This will cause pushing a new stack frame to the execution context and changing the graph
-    /// controller to point to a new definition.
+    /// This will push new stack frames to the execution context and change the graph controller to
+    /// point to a new definition.
     ///
     /// ### Errors
     /// - Fails if method graph cannot be created (see `graph_for_method` documentation).
     /// - Fails if the project is in read-only mode.
-    pub async fn enter_method_pointer(&self, local_call: &LocalCall) -> FallibleResult {
+    pub async fn enter_stack(&self, stack: Vec<LocalCall>) -> FallibleResult {
```

Comment:
I don't think so. We need an owned version both for pushing them to the stack in the model and for the notification back to the view.

In case you already realised that and were wondering about doing an extra `clone` here instead of at the call site, I'm trying to follow [this pattern](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control) which helps make it clear when cloning is happening.

Reply:
If we definitely need an owned version of vector inside, and we would be cloning it nevertheless if was provided as a slice, then yes, API telling explicitly about it is better.